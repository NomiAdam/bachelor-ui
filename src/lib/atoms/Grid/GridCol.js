import PropTypes from 'prop-types';
import styled from 'styled-components';
import device from '../../constants/device';

const StyledGridCol = styled.div`
	background-color: transparent;
	${ ({ center }) => (center ? `
	display: flex;
	justify-content: center;
	align-items: center;
	` : '') }
	grid-column-end: span ${ props => props.colXS || props.colMD || props.colLG || 12 };
	@media ${ device.mobileS } {
		grid-column-end: span ${ props => props.colXS };
	}
	@media ${ device.mobileM } {
		grid-column-end: span ${ props => props.colXS };
	}
	@media ${ device.mobileL } {
		grid-column-end: span ${ props => props.colXS };
	}
	@media ${ device.tablet } {
		grid-column-end: span ${ props => props.colMD || props.colXS };
	}
	@media ${ device.laptop } {
		grid-column-end: span ${ props => props.colMD || props.colXS };
	}
	@media ${ device.laptopL } {
		grid-column-end: span ${ props => props.colLG || props.colMD || props.colXS };
	}	
	@media ${ device.desktop } {
		grid-column-end: span ${ props => props.colLG || props.colMD || props.colXS };
	}
	@media ${ device.desktopL } {
		grid-column-end: span ${ props => props.colLG || props.colMD || props.colXS };
	}
`;

StyledGridCol.propTypes = {
    /**
	 * Number representing how many column will GridCol span from 0-12 on small devices
	 */
    colXS: PropTypes.number,
    /**
	 * Number representing how many column will GridCol span from 0-12 on medium devices
	 */
    colMD: PropTypes.number,
    /**
	 * Number representing how many column will GridCol span from 0-12 on large devices
	 */
    colLG: PropTypes.number,
    /**
	 * Whether we should center context in GridCol or not
     */
    center: PropTypes.bool,
};

export default StyledGridCol;
