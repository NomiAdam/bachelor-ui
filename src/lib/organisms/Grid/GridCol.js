import PropTypes from 'prop-types';
import styled from 'styled-components';
import device from '../../constants/device';

const GridCol = styled.div`
	background-color: transparent;
	display: flex;
	justify-content: ${ ({ horizontalAlign }) => horizontalAlign };
	align-items: ${ ({ verticalAlign }) => verticalAlign };
	grid-column-end: span ${ props => props.colXS || props.colMD || props.colLG || 12 };
	@media ${ device.mobileS } {
		grid-column-end: span ${ props => props.colXS };
	}
	@media ${ device.mobileM } {
		grid-column-end: span ${ props => props.colXS };
	}
	@media ${ device.mobileL } {
		grid-column-end: span ${ props => props.colXS };
	}
	@media ${ device.tablet } {
		grid-column-end: span ${ props => props.colMD || props.colXS };
	}
	@media ${ device.laptop } {
		grid-column-end: span ${ props => props.colMD || props.colXS };
	}
	@media ${ device.laptopL } {
		grid-column-end: span ${ props => props.colLG || props.colMD || props.colXS };
	}	
	@media ${ device.desktop } {
		grid-column-end: span ${ props => props.colLG || props.colMD || props.colXS };
	}
	@media ${ device.desktopL } {
		grid-column-end: span ${ props => props.colLG || props.colMD || props.colXS };
	}
`;

GridCol.propTypes = {
    /**
	 * Number representing how many column will GridCol span from 0-12 on small devices
	 */
    colXS: PropTypes.number,
    /**
	 * Number representing how many column will GridCol span from 0-12 on medium devices
	 */
    colMD: PropTypes.number,
    /**
	 * Number representing how many column will GridCol span from 0-12 on large devices
	 */
    colLG: PropTypes.number,
    /**
	 * String representation of horizontal alignment in flex box
     */
    horizontalAlign: PropTypes.string,
    /**
   * String representation of horizontal alignment in flex box
   */
    verticalAlign: PropTypes.string,
};

GridCol.defaultProps = {
    horizontalAlign: 'flex-start',
    verticalAlign: 'flex-start',
};

export default GridCol;
