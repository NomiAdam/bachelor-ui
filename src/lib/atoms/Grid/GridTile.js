import React from 'react';
import PropTypes from 'prop-types';
import Paper from '../Paper/index';
import GridCol from './GridCol';
import { lightTheme } from '../../constants/theme';

const GridTile = ({ background, width, height, children, padding, center, ...rest }) => (
    <GridCol {...rest} center={center}>
        <Paper background={background} width={width} height={height} padding={padding} center={center} >
            {children}
        </Paper>
    </GridCol>
);

GridTile.propTypes = {
    /**
     * String representation of background color
     */
    background: PropTypes.string,
    /**
     * String representation of width
     */
    width: PropTypes.string,
    /**
     * String representation of height
     */
    height: PropTypes.string,
    /**
     * Whether we should center inner parts
     */
    center: PropTypes.bool,
};

GridTile.defaultProps = {
    width: '100%',
    background: lightTheme.white,
};

export default GridTile;
