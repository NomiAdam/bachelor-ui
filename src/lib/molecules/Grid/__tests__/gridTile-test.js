/* eslint-disable no-undef */
import React from 'react';
import GridTile from '../GridTile';
import GridCol from '../GridCol';
import { shallow } from 'enzyme';

describe('Checkbox', () => {
    it('should be defined', () => {
        expect(GridTile).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<GridTile />)).toMatchSnapshot();
    });
    it('should match be of type GridCol', () => {
        expect(shallow(<GridTile />).type()).toBe(GridCol);
    });
});
