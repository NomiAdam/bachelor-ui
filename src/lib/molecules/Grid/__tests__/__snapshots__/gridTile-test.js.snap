// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match the snapshot 1`] = `
<styled.div
  horizontalAlign="flex-start"
  verticalAlign="flex-start"
>
  <Paper
    background="#ffffff"
    cursor="initial"
    margin="0"
    overflow="hidden"
    padding="10px"
    width="100%"
  />
</styled.div>
`;
